pandas/core/strings.py

```diff
@@ -1020,7 +1024,17 @@ def str_extractall(arr, pat, flags=0):
 
     index = MultiIndex.from_tuples(index_list, names=arr.index.names + ["match"])
 
-    result = arr._constructor_expanddim(match_list, index=index, columns=columns)
+    # workaround #27953
+    # ideally we just pass `dtype=arr.dtype` unconditionally, but this fails
+    # when the list of values is empty.
+    if arr.dtype.name == "text":
+        dtype = arr.dtype
+    else:
+        dtype = None
```

Comment:
shouldn't this by dtype=object and then maybe can just re-use the codepath in _str_extract_frame (maybe make a helper function)